core/eolearn/core/eodata_io.py

```diff
@@ -401,17 +402,44 @@ def load(self) -> T:
         """Method for loading a feature. The loaded value is stored into an attribute in case a second load request is
         triggered inside a shallow-copied EOPatch.
         """
-        if self.loaded_value is not None:
-            return self.loaded_value
+        if self.loaded_value is None:
+            self.loaded_value = self._load_value()
+
+        return self.loaded_value
+
+    @abstractmethod
+    def _load_value(self) -> T:
+        """Loads the value from the storage."""
+
+    @classmethod
+    @abstractmethod
+    def save(cls, data: T, filesystem: FS, feature_path: str, compress_level: int = 0) -> None:
+        """Method for saving a feature. The path is assumed to be filesystem path but without file extensions.
+
+        Example of path is `eopatch/data/NDVI`, which is then used to save `eopatch/data/NDVI.npy.gz`.
+        """
+
+
+class FeatureIOGZip(FeatureIO[T], metaclass=ABCMeta):
```

Comment:
I dont want the Base prefix because it is used as a type annotation. The GZip suffix was added since the class adds functionality for using GZip which is then used by all the other subclasses. Zarr cannot be used together with GZip and will thus only inherit directly from FeatureIO.

I find it more correct to not hardcode GZip into the base class if it's not used by all subclasses.

Reply:
discussed privately, resolving